### PR TITLE
Pass full command name to executables

### PR DESCRIPTION
**Pull request by jdevera**
Use the ${$PREFIX}_COMMAND_NAME environment variable to pass the full command to the executable files when they are run.

The full command would be the app name, followed optionally by the group name, and finally the command name that was configured in the manifest to execute the command.

This can be used, for instance, by programs to output a meaningful usage message that refers to how the program was actually executed.

### CODE REVIEW

File: internal/context/context.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ type LauncherContext interface {
 
 	CmdPackageDirEnvVar() string
 
+	CmdNameEnvVar() string
```
Comment (bhou) [inline internal/context/context.go:30]: Same here, FullCmdNameEnvVar()

File: internal/frontend/default-frontend.go [hunk 2]
```diff
@@ -261,6 +261,15 @@ func formatExamples(examples []command.ExampleEntry) string {
 	return strings.Join(output, "\n")
 }
 
+func (self *defaultFrontend) getCommandName(group, name string) string {
```
Comment (bhou) [inline internal/frontend/default-frontend.go:264]: getFullCommandName()

File: internal/frontend/default-frontend.go [hunk 3]
```diff
@@ -261,6 +261,15 @@ func formatExamples(examples []command.ExampleEntry) string {
...
+	tokens := []string{self.appCtx.AppName()}
```
Comment (bhou) [inline internal/frontend/default-frontend.go:265]: This appName returned here is hard coded in the main.go file. This is used for the command launcher to get the environment variable PREFIX and determine the default configuration locations: `$HOME/.${appName}`

The code here might not cover a situation in which the user renames the binary into something else. For example, the hard-coded binary is "cl", the user could rename it to "mycl". In this case, the correct value for `CL_FULL_COMMAND_NAME` will be `mycl [group] [name]`.

File: test/README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ To run all the integration test simply run the script: `integration.sh`
 
 ```bash
 cd command-launcher
-./test/interation.sh
+./test/integration.sh
```
Comment (bhou) [inline test/README.md:7]: Nice catch :D

**Review by bhou — CHANGES_REQUESTED** [review]
Please rename the command to avoid ambiguity, and handle the situations that user renamed the binary itself.

File: test/packages-src/bonjour/bonjour.sh [hunk 5]
```diff
@@ -7,6 +7,7 @@ echo $CL_FLAG_LANGUAGE
 echo $CL_ARG_1
 
 echo "packge dir: $CL_PACKAGE_DIR"
+echo "command name: $CL_COMMAND_NAME"
```
Comment (bhou) [inline test/packages-src/bonjour/bonjour.sh:10]: Need to modify the bonjour.bat file as well to cover windows cases

File: internal/frontend/default-frontend.go (hunk 3 above)
Comment (jdevera) [reply internal/frontend/default-frontend.go]: Good point, I change that to use the actual executable name.

**Comment by jdevera** [comment]
Thanks for the review!

In my second round of changes, I:

* Renamed the variable to `${PREFIX}_FULL_COMMAND_NAME`
* Loaded it with the actual name of the executable for the launcher (still with a fallback to the configured name at build time in case of errors)
* Added the variable to the .bat file to fix windows tests
* Changed the test to run on a renamed binary, so it also checks that the actual name is used

File: test/packages-src/bonjour/bonjour.bat [hunk 6]
```diff
@@ -4,6 +4,7 @@ ECHO %CL_FLAG_LANGUAGE%
 ECHO %CL_ARG_1%
 
 ECHO packge dir: %CL_PACKAGE_DIR%
+ECHO command name: %CL_COMMAND_NAME%
```
Comment (bhou) [inline test/packages-src/bonjour/bonjour.bat:7]: This shoule be %CL_FULL_COMMAND_NAME%. It fails the integration tests.

**Review by bhou — CHANGES_REQUESTED** [review]
The environment variable name in .bat script is not correct, which fails the integration test

File: test/packages-src/bonjour/bonjour.bat (hunk 6 above)
Comment (jdevera) [reply test/packages-src/bonjour/bonjour.bat]: 🤦 of course, my bad. Sorry for the extra back and forth, the good news is that after a first contribution, I can see the test results and iterate before you waste time 😅 .

**Review by bhou — APPROVED** [review]

**Pull request merged**